ProjFS.Mac/PrjFSKextTests/UtilitiesTests.m

```diff
@@ -0,0 +1,18 @@
+#import <XCTest/XCTest.h>
+#include "../PrjFSKext/Utilities.hpp"
+
+@interface UtilitiesTests : XCTestCase
+
+@end
+
+@implementation UtilitiesTests
+
+- (void)testActionBitIsSet {
+    XCTAssertTrue(ActionBitIsSet(1,1));
+    XCTAssertTrue(ActionBitIsSet(3,1));
+    XCTAssertTrue(ActionBitIsSet(5,1));
+    XCTAssertFalse(ActionBitIsSet(6,1));
+    XCTAssertTrue(ActionBitIsSet(6,2));
```

Comment:
I realise the function under test is extremely basic and so this is more of a demo, but is there any particular reason you're just using integer literals rather than the symbolic `KAUTH_VNODE_*` constants it would actually be used with in practice? `#include <sys/kauth.h>` in userspace seems to work just fine and makes those constants available in the test method. It would be nice if the tests were at least loosely based in reality.

Reply:
I was debating if we should keep to primitive types / mocks, but I agree the test looks old.  Further to that we use Utilities which includes <sys/kernel_types.h> which pulls in the constants anyways.  I've updated the test to bemore realistic, thanks!